metadata-service/factories/src/main/java/com/linkedin/gms/factory/auth/AuthorizerChainFactory.java

```diff
@@ -65,53 +80,67 @@ protected AuthorizerChain getInstance() {
-          log.info(String.format("Authorizer %s is initialized", type));
-        } catch (Exception e) {
-          throw new RuntimeException(
-              String.format("Failed to instantiate custom Authorizer with class name %s", clazz.getCanonicalName()), e);
-        }
-      }
-    }
+    Path pluginBaseDirectory = Paths.get(configurationProvider.getDatahub().getPlugin().getAuth().getPath());
+    ConfigProvider configProvider =
+        new ConfigProvider(pluginBaseDirectory);
+
+    Optional<Config> optionalConfig = configProvider.load();
+    // Register authorizer plugins if present
+    optionalConfig.ifPresent((config) -> {
+      registerAuthorizer(customAuthorizers, resolver, config);
+    });
+
     return customAuthorizers;
   }
+
+  private void registerAuthorizer(List<Authorizer> customAuthorizers, ResourceSpecResolver resolver, Config config) {
+    PluginConfigFactory<AuthorizerPluginConfig> authorizerPluginPluginConfigFactory = new PluginConfigFactory<>(config);
+    // Load only Authorizer configuration from plugin config factory
+    List<AuthorizerPluginConfig> authorizers =
+        authorizerPluginPluginConfigFactory.loadPluginConfigs(PluginType.AUTHORIZER);
+
+    // Select only enabled authorizer for instantiation
+    List<AuthorizerPluginConfig> enabledAuthorizers = authorizers.stream().filter(pluginConfig -> {
+      if (!pluginConfig.getEnabled()) {
+        log.info(String.format("Authorizer %s is not enabled", pluginConfig.getName()));
+      }
+      return pluginConfig.getEnabled();
+    }).collect(Collectors.toList());
+
+    // Get security mode set by user
+    SecurityMode securityMode =
+        SecurityMode.valueOf(this.configurationProvider.getDatahub().getPlugin().getPluginSecurityMode());
+    // Create permission manager with security mode
+    PluginPermissionManager permissionManager = new PluginPermissionManagerImpl(securityMode);
+
+    // Save ContextClassLoader. As some plugins are directly using context classloader from current thread to load libraries
+    // This will break plugin as their dependencies are inside plugin directory only
+    ClassLoader contextClassLoader = Thread.currentThread().getContextClassLoader();
+    // Instantiate Authorizer plugins
+    enabledAuthorizers.forEach((pluginConfig) -> {
+      // Create context
+      AuthorizerContext context = new AuthorizerContext(
+          ImmutableMap.of(PluginConstant.PLUGIN_HOME, pluginConfig.getPluginHomeDirectory().toString()), resolver);
+      IsolatedClassLoader isolatedClassLoader = new IsolatedClassLoaderImpl(permissionManager, pluginConfig);
+      try {
+        Thread.currentThread().setContextClassLoader((ClassLoader) isolatedClassLoader);
+        Authorizer authorizer = (Authorizer) isolatedClassLoader.instantiatePlugin(Authorizer.class);
+        log.info("Initializing plugin {}", pluginConfig.getName());
+        authorizer.init(pluginConfig.getConfigs().orElse(Collections.emptyMap()), context);
```

Comment:
What if _init_ throws - for example the configs are wrong. 

We should try-catch and log a useful message to the user

Reply:
please ^